### PR TITLE
UIU-3214 - Review and cleanup Module Descriptors for ui-users

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
[UIU-3214](https://folio-org.atlassian.net/browse/UIU-3214) - Review and cleanup Module Descriptors for ui-users

## Approach
30 invalid permission names have been updated.

#### TODOS and Open Questions
image

2/30 permissions, namely "Users: User loans: add patron information"(ui-users.loans.add-patron-info) and "Users: User loans: add staff information"(ui-users.loans.add-staff-info) are composed of same sub permissions. So, I have created a new permission "Users: User loans: add staff or patron information"(ui-users.loans.add-info.create) that replaces the two old ones. 
If you see any objections in this approach, please let me know.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -936,41 +952,46 @@
...
+        "replaces": ["ui-users.overrideItemBlock"],
         "subPermissions": [
           "circulation.override-item-limit-block",
           "circulation.override-item-not-loanable-block"
         ],
         "visible": true
       },
       {
-        "permissionName": "ui-users.cashDrawerReport",
+        "permissionName": "ui-users.cashDrawerReport.execute",
```
Comment (mariia-aloshyna) [inline package.json:974]:
```suggestion
        "permissionName": "ui-users.cash-drawer-report.create",
```

File: package.json [hunk 2]
```diff
@@ -936,41 +952,46 @@
...
         "displayName": "Users: Create and download Cash drawer reconciliation report",
+        "replaces": ["ui-users.cashDrawerReport"],
         "subPermissions": [
           "feefine-reports.cash-drawer-reconciliation.post"
         ],
         "visible": true
       },
       {
-        "permissionName": "ui-users.financialTransactionReport",
+        "permissionName": "ui-users.financialTransactionReport.execute",
```
Comment (mariia-aloshyna) [inline package.json:983]:
```suggestion
        "permissionName": "ui-users.financial-transaction-report.create",
```

File: package.json [hunk 3]
```diff
@@ -475,8 +482,9 @@
         "visible": true
       },
       {
-        "permissionName": "ui-users.accounts",
+        "permissionName": "ui-users.accounts.all",
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply package.json:485]: Probably, I didn't get your question correctly. 
`displayName` clearly states that this permission allows the user to create, edit and remove accounts. Owing to [General Permissions](https://folio-org.atlassian.net/wiki/spaces/FOLIJET/pages/156368925/Permissions+naming+convention#7.-General-Permissions) mention on Permissions naming convention page, I named it as `all`

File: package.json [hunk 4]
```diff
@@ -494,8 +502,9 @@
         "visible": true
       },
       {
-        "permissionName": "ui-users.feefineactions",
+        "permissionName": "ui-users.feefineactions.all",
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply package.json:505]: > same as above :)

File: package.json [hunk 5]
```diff
@@ -528,8 +537,9 @@
         "visible": true
       },
       {
-        "permissionName": "ui-users.manual_pay",
+        "permissionName": "ui-users.manual_pay.execute",
```
[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply package.json:540]: here, I retained '_' as it is defined earlier. For the sake of consistency, I will update this with '-' along with other permissions.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (mariia-aloshyna) [reply package.json:485]: I mean you have almost the same `Users: Can create, edit and remove proxies` and you changed it to `ui-users.proxies.edit`, but here you used `all`, so what is the difference?

[the remark this answers is not in the document]
Comment (Terala-Priyanka) [reply package.json:485]: Hey Mariia,
I was keen on the previous permissionName - `"ui-users.editproxies"`
Thanks for catching it. I will update it with `all`

**Review by mariia-aloshyna — APPROVED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Pull request merged**